src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLFunctionsIT.java

```diff
@@ -242,6 +236,93 @@ public void functionPow() throws Exception {
         );
     }
 
+    @Test
+    public void operatorSubstring() {
+        assertThat(
+                executeQuery("SELECT substring('sampleName', 1, 4) FROM " + TEST_INDEX_ACCOUNT,
+                        "jdbc"),
+                containsString("samp")
+        );
+
+        assertThat(
+                executeQuery("SELECT substring('sampleName', 0, 20) FROM " + TEST_INDEX_ACCOUNT,
+                        "jdbc"),
+                containsString("sampleName")
+        );
+    }
+
+    @Test
+    public void operatorLength() throws IOException {
+        assertThat(
+                executeQuery("SELECT LENGTH(lastname) FROM " + TEST_INDEX_ACCOUNT
+                                + " WHERE lastname IS NOT NULL GROUP BY LENGTH(lastname) ORDER BY LENGTH(lastname)", "jdbc"),
+                containsString("\"type\": \"integer\"")
+        );
+
+        assertThat(
+                executeQuery("SELECT LENGTH('sampleName') AS length FROM " + TEST_INDEX_ACCOUNT),
+                hitAny(kvInt("/fields/length/0", equalTo(10)))
+        );
+
+    }
+
+    @Test
+    public void operatorReplace() {
+        String query = "SELECT REPLACE('elastic', 'el', 'fant') FROM " + TEST_INDEX_ACCOUNT;
+        assertThat(
+                executeQuery(query, "jdbc"),
+                containsString("fantastic")
+        );
+    }
+
+    @Test
+    public void operatorLocate() throws IOException {
+        String query = "SELECT LOCATE('a', lastname, 0) FROM " + TEST_INDEX_ACCOUNT
+                + " WHERE lastname IS NOT NULL GROUP BY LOCATE('a', lastname, 0) ORDER BY LOCATE('a', lastname, 0)";
+        assertThat(
+                executeQuery(query, "jdbc"), containsString("\"type\": \"integer\"")
+        );
+
+        assertThat(
+                executeQuery("SELECT LOCATE('a', 'sampleName', 3) AS locate FROM " + TEST_INDEX_ACCOUNT),
+                hitAny(kvInt("/fields/locate/0", equalTo(8)))
+        );
+        assertThat(
+                executeQuery("SELECT LOCATE('a', 'sampleName') AS locate FROM " + TEST_INDEX_ACCOUNT),
+                hitAny(kvInt("/fields/locate/0", equalTo(2)))
+        );
+    }
+
+    @Test
+    public void rtrim() {
+        assertThat(
+                executeQuery("SELECT RTRIM(' sampleName  ') FROM " + TEST_INDEX_ACCOUNT, "jdbc"),
+                containsString(" sampleName")
+        );
+    }
+
+    @Test
+    public void ltrim() {
+        assertThat(
+                executeQuery("SELECT LTRIM(' sampleName  ') FROM " + TEST_INDEX_ACCOUNT, "jdbc"),
+                containsString("sampleName  ")
```

Comment:
Here also add a negative assert to check it does not contain ` "   sampleName "`, because `containsString` assertion will work even if it not left trimmed.

Reply:
@abbashus I've changed the assert in `substring()`, `ltrim()` and `rtrim()` to compare the content in /hits/fields array of the response and that would be more straightforward. Thanks!